spec/statement.md

```diff
@@ -0,0 +1,174 @@
+> end blocks of statements.
+
+[`documentation indentation character`]: #document-indentation-character
+
+### Document indentation character
+
+The `document indentation character` is defined as the first tab or space
+character used in an [`IndentMore`] production in a document.
+
+[`IndentMore`]: #indentation
+
+> The `document indentation character` is the character used for calculating the
+> [`current indentation level`]. In the indented syntax, no character other than
+> the `document indentation character` may be used for indentation.
+
+[`current indentation level`]: #current-indentation-level
+
+## Syntax
+
+### ScssStatements
+
+<x><pre>
+**ScssStatements**      ::= (Statement ';'?¹)* Statement?
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment`, `SilentComment`, or ends in a `Block`.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement [IndentSame])* Statement
+</pre></x>
+
+[IndentSame]: #indentation
+
+The `Statement` productions may not include newlines outside of `IndentSame`
+productions.
+
+### Stylesheet
+
+<x><pre>
+**Stylesheet**          ::= U+FEFF? ([ScssStatements] | [IndentedStatements])¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+1: Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements]
+**Block**          ::= (ScssBlock | IndentedBlock)¹
+</pre></x>
+
+[IndentMore]: #indentation
+
+1: Only the production for the current syntax is valid.
+
+### Comments
+
+#### LoudComment
+
+<x><pre>
+**ScssLoudComment**          ::= '/\*' (.\*¹ | Interpolation)\* '\*/'
+**InterpolatedCommentText**  ::= (.\*² | Interpolation)\*
+**IndentedLoudChildren**     ::= (InterpolatedCommentText [IndentSame])\*
+&#32;                            InterpolatedCommentText
+**IndentedLoudComment**      ::= '/\*' InterpolatedCommentText
+&#32;                            ([IndentMore] IndentedLoudChildren)?
+**LoudComment**              ::= (ScssLoudComment | IndentedLoudComment)³
+</pre></x>
+
+1. This may not contain `#{` or `*/`.
+2. This may not contain `#{` or newlines.
+3. Only the production for the current syntax is valid.
+
+#### SilentComment
+
+<x><pre>
+**ScssSilentComment**          ::= '//' .\*¹
+**CommentText**                ::= .\*²
+**IndentedSilentChildren**     ::= (CommentText [IndentSame])\* CommentText
+**IndentedSilentComment**      ::= '//' CommentText ([IndentMore]
+&#32;                              IndentedSilentChildren)?
+**SilentComment**              ::= (ScssSilentComment | IndentedSilentComment)³
+</pre></x>
+
+1. This may not contain newlines.
+2. This may not contain newlines outside of [IndentSame] productions.
```

Comment:
Good point. I changed it to `[WhitespaceOnlyLine]` because [silent comments consume](https://sass-lang.com/playground/#eJwVizEOwjAQBPt7xZZQBdr8xlwWfJJzjuyLIL/HLqYYaebxXBYRIA2A1wqdFpnQuu/0gCaHuZZzI77Zgv1ISlQvF4o5++hvc+iXR/oh2yeXQbDB+lxra9S4i8g2Wq54/wEWeCSu) WhitespaceOnlyLines as well.

Reply:
The issue is that this doesn't contain *any* other productions. It's just matching pure characters. It should forbid newlines entirely.

Reply:
Ok, that makes sense. I forbade newlines in https://github.com/sass/sass/pull/3969/commits/1c19f21c9e0a9c5736dfc78d52839c10480af847.